pulser-core/pulser/json/abstract_repr/serializer.py

```diff
@@ -230,10 +243,19 @@ def remove_kwarg_if_default(
             )
         elif "target" in call.name:
             data = get_all_args(("qubits", "channel"), call)
+            target: Parametrized | int | list[int]
             if call.name == "target":
                 target = convert_targets(data["qubits"])
             elif call.name == "target_index":
-                target = data["qubits"]
+                if isinstance(
+                    data["qubits"], pulser.parametrized.Parametrized
+                ):
+                    # The qubits are given as a variable
+                    target = data["qubits"]
+                else:
+                    target = cast(
```

Comment:
@HGSilveri can you add a comment about the possible target types (like done above)?